libcontainer/cgroups/file.go

```
@@ -171,9 +171,7 @@ func openFile(dir, file string, flags int) (*os.File, error) {
 		//
 		// TODO: if such usage will ever be common, amend this
 		// to reopen cgroupRootHandle and retry openat2.
-		fdPath, closer := utils.ProcThreadSelf("fd/" + strconv.Itoa(int(cgroupRootHandle.Fd())))
-		defer closer()
-		fdDest, _ := os.Readlink(fdPath)
+		fdDest, _ := os.Readlink("/proc/thread-self/fd/" + strconv.Itoa(int(cgroupRootHandle.Fd())))
```

Maybe there are some way to make `/proc/thread-self` inaccessible, for example:
1. Some mount options for proc fs;
2. Kernel customization;
3. Custom selinux rules.

So I think we still need to fallback to `/proc/self/task/$TID` when /proc/thread-self is inaccessible, but I think it's not very important here, because this is an error message enhancement, maybe we can change the code like this:
```go
if fdDest, err1 := os.Readlink("/proc/thread-self/fd/" + strconv.Itoa(int(cgroupRootHandle.Fd()))); err1 == nil && fdDest != cgroupfsDir {
    // Wrap the error so it is clear that cgroupRootHandle
    // is opened to an unexpected/wrong directory.
    err = fmt.Errorf("cgroupRootHandle %d unexpectedly opened to %s != %s: %w",
		cgroupRootHandle.Fd(), fdDest, cgroupfsDir, err)
}
```
Or add a check `fdDest != ""` in the next if.

---

This is not really related to the patch, as we used to ignore the error before, too.

With the current code, we can guess that readlink failed from the error message (which would be something like `cgroupRootHandle 9 unexpectedly opened to != /sys/fs/cgroup: openat2...`). It's not ideal, but it's better than nothing because we see something is wrong.

If we do check the readlink error (or add a check for `fdDest ~= ""`), we won't see this at all, we'll just have "openat2 failed".

Anyway, updated to check for the os.Readlink error.